app/services/adapters/web-extension.js

```diff
@@ -60,12 +60,23 @@ export default class WebExtension extends BasicAdapter {
   }
 
   _connect() {
+    if (this._isConnected) {
+      return;
+    }
     let chromePort = this._chromePort;
     chromePort.postMessage({ appId: chrome.devtools.inspectedWindow.tabId });
 
     chromePort.onMessage.addListener((...args) => {
       this._messageReceived(...args);
     });
+
+    chromePort.onDisconnect.addListener(() => {
+      this._isConnected = false;
```

Comment:
Do we need this `this._isConnected` boolean? If we're going to call `this._connect` `onDisconnect` wouldn't that be sufficient?

Reply:
We don't. fixed